packages/atomic/src/utils/local-storage-utils.tsx

```diff
@@ -0,0 +1,108 @@
+import {SearchEngine} from '@coveo/headless';
+
+export enum StorageItems {
+  RECENT_QUERIES = 'coveo-recent-queries',
+}
+
+export class SafeStorage implements Storage {
+  constructor(private logger: SearchEngine['logger']) {}
+  public clear() {
+    return this.tryAccessLocalStorageOrWarn(
+      () => localStorage.clear(),
+      () => {}
+    );
+  }
+
+  public getItem(key: StorageItems) {
+    return this.tryAccessLocalStorageOrWarn(
+      () => localStorage.getItem(key),
+      () => null
+    );
+  }
+
+  public getItemParsed<T>(key: StorageItems, fallback: T) {
+    const item = this.getItem(key);
+    if (!item) {
+      return fallback;
+    }
+    return this.tryJSONOrWarn(
+      key,
+      () => JSON.parse(item) as T,
+      () => fallback
+    );
+  }
+
+  public key(index: number) {
+    return this.tryAccessLocalStorageOrWarn(
+      () => localStorage.key(index),
+      () => null
+    );
+  }
+
+  public get length() {
+    return this.tryOrElse(
+      () => localStorage.length,
+      () => 0
+    );
+  }
+
+  public removeItem(key: StorageItems) {
+    return this.tryAccessLocalStorageOrWarn(
+      () => localStorage.removeItem(key),
+      () => {}
+    );
+  }
+
+  public setItem(key: StorageItems, value: string) {
+    return this.tryAccessLocalStorageOrWarn(
+      () => localStorage.setItem(key, value),
+      () => {}
+    );
+  }
+
+  public setObject(key: StorageItems, obj: any) {
```

Comment:
Instead of `getItemParsed` and `setObject`, could `setJSON` and `getParsedJSON` be a more aligned pair of names?